applications/ShapeOptimizationApplication/python_scripts/algorithm_factory.py

```diff
@@ -26,6 +30,13 @@ def CreateOptimizationAlgorithm(optimization_settings, analyzer, communicator, m
                                         analyzer,
                                         communicator,
                                         model_part_controller)
+    elif algorithm_name == "gradient_projection":
+        if not AlgorithmGradientProjection:
+            raise RuntimeError("'gradient_projection' algorithm is not available. Hint: 'EigenSolversApplication' is required.")
+        return AlgorithmGradientProjection(optimization_settings,
```

Comment:
I don't think this is a good solution

that why e.g. in [StructuralMechanics](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/python_solvers_wrapper_structural.py) in the solver-factory we only import the things we really need

Reply:
We have actually many places in the ShapeOpt app that deal with imports that may not be available in this fashion. I think we had the imports inside the if/else statements at some points but moved them to the top of the file because of codacy/pep8.

What is the advantage in doing like in the StructMech, other then avoiding this strange trick with `if not AlgorithmGradientProjection:`?

I can adapt it, but i would like to make sure what i plan doing is ok:

```
if algorithm_name == "gradient_projection":
    from .algorithm_gradient_projection import AlgorithmGradientProjection
    return AlgorithmGradientProjection(optimization_settings, ...)
```

But to be consistent i would use this pattern for all other algorithms that are always available as well.

Reply:
> What is the advantage in doing like in the StructMech, other then avoiding this strange trick with if not AlgorithmGradientProjection:?

I think this is to make it easier for the user, e.g. for serial/MPI selection

To me your solution looks good! TBH I don't really know why structural doesn't do it in this way

Reply:
I see, there some runtime string manipulation is done to find the exact module that should be imported. Here the mechanism is simpler. So if you don't mind i would keep it as is.